[READY] Blob Mania 1: Reagent Blobs, Split Personality, Storage Blobs, and Blob Broadcast

Iamgoofball opened the pull request:
![hrwelec](https://cloud.githubusercontent.com/assets/4081722/5698280/1c55630a-99bf-11e4-913b-54d980543d49.png)

Blobs now start with a random blob specific reagent.
You can easily add new ones by adding reagents to Blob-Reagents.dm with the path /datum/reagent/blob/reagent_name_here
Blobbernauts now inflict the reagent that the blob starts with instead of "lol infinite burn"
Blobs inflict their reagent on mobs upon expanding onto a tile.
When blobs talk to other blobs over telepathy, it shows their reagent.
Adds "Split consciousness" a 100 cost, node requiring (think blobbernaut) power that produces a second blob overmind (with it's own unique reagent!)
Adds Storage Blobs. Each blob gives you 50 extra max points.
Adds Blob Broadcast.  It's a free verb that will make all your blob spores, zombies, and blobbernauts speak for you. They'll say whatever you want to say, and they'll all say it at once.

This should change up the blob metagame a bit, and allows for more opportunities for blobs to work together on stuff.

Let's say the crew's heavily armored against brute damage and has welders. Have your Acidic Liquid blob buddy sneak in nearby and eliminate their armor while you finish them off with your increased brute damage Skin Ripper reagent.

Iamgoofball committed:
Reagent Blobs

Iamgoofball committed:
whoops

Iamgoofball committed:
forgot some things for acid blob

Razharas commented:
Oh wow, now blobs and vines are even closer

Iamgoofball commented:
:^)

Iamgoofball committed:
makes the code compile, fixes acid values

Steelpoint commented:
So this is a random thing you cannot control?

On the one hand that stops everyone for going to the "meta" choice, on the other hand its pretty random.

RemieRichards commented on code/game/gamemodes/blob/blobs/blob_mobs.dm:
```diff
@@ -132,14 +132,13 @@
-		var/mob/living/L = target
-		if(L.reagents)
-			L.reagents.add_reagent("spore_burning", 10)
+		overmind.blob_reagent_datum.reaction_mob(target, TOUCH)
```
you should add a check that the overmind exists.

Razharas commented:
Stamina blob a shite

RemieRichards commented on code/game/gamemodes/blob/overmind.dm:
```diff
@@ -15,18 +15,30 @@
 	var/last_attack = 0
+	var/datum/reagent/blob_reagent = "water"
+	var/datum/reagent/blob_reagent_name = "Water"
+	var/datum/reagent/blob_color = ""
```
you only need the blob_reagent_datum var, the rest can be obtained using that without separate vars.

RemieRichards commented on code/game/gamemodes/blob/powers.dm:
```diff
@@ -265,6 +269,10 @@
 		return
 	last_attack = world.time
 	OB.expand(T, 0)
+	for(var/atom/movable/A in T)
```
why not just do ```for(var/mob/living/L in T) blob_reagent_datum.reaction_mob(L, TOUCH)```

Razharas commented:
Things to concider:
* Does brute damage get reduced by armour? If yes then brute blob too weak since other things dont
* Maybe adjust bliblings and blobernaughts to the type of blob?
* Pure stamina damage is just too bad
* Different types dont really change much of anything since apart from brute other damage doesnt get cut by armour
* Those that do act differently namely acid are too OP since they do same damage but also kill items on you

RemieRichards committed:
Merge branch 'raginmages' of https://github.com/Iamgoofball/-tg-station into GoofBlob

Iamgoofball commented:
They are greyscaled. It looks better on other colors.
I'll adjust bruteblob.
Blobbernauts are adjusted, it's their nerf.
I'll add some damage stuff to stamina blob.
Different damage types do have effects, toxloss can stop you in important situations to throw up, oxyloss can speed up the crit process and prevent you being saved, and burnloss I think can light you on fire iunno
Acid is supposed to kill items, don't get up close and personal with acid blob. I'll lower it's base brute damage output though.

Iamgoofball commented:
They are greyscaled. It looks better on other colors.
I'll adjust bruteblob.
Blobbernauts are adjusted, it's their nerf.
I'll add some damage stuff to stamina blob.
Different damage types do have effects, toxloss can stop you in important situations to throw up, oxyloss can speed up the crit process and prevent you being saved, and burnloss I think can light you on fire iunno
Acid is supposed to kill items, don't get up close and personal with acid blob. I'll lower it's base brute damage output though.

Iamgoofball commented:
whoops

Razharas commented:
Make core and node icons be black maybe or something
Currently they blend in with usual blob tiles too much

Razharas commented:
Burn damage cant light you on fire, lighting on fire is special snowflake

Razharas commented:
By adjusting bloblings and blobbernaughts i mostly meant their damage type

RemieRichards committed:
Adds "Split consciousness" a 100 cost, node requiring (think blobbernaut) power that produces a second blob overmind (with it's own unique reagent!)
Alters code to require only 1 var instead of 4, alters some cases where blobs of a different colour change.
Fixes the core and shield blobs of the overmind sometimes being a different colour to the rest.
Makes blob spores the same colour as their overmind, but retain the husk icon in zombie mode (These are slightly off colour due to still being orange by default)

RemieRichards commented:
@Iamgoofball https://github.com/Iamgoofball/-tg-station/pull/3
I fixed some things I complained about and added the new power discussed in IRC.

WJohn commented:
While I love the idea, I'm not hugely fond of the 255 red I'm looking at at the picture. I'm hoping you'll tone it down or have some different shades like you did with your cooking PR.

Iamgoofball committed:
Merge pull request #3 from RemieRichards/GoofBlob

Assorted blob changes

paprka commented:
The color idea is good and should use the 'alien color' selection like lizards do, but this is a terrible approach to countering riot armor opponents, worse than blobbernauts doing the infinite flame thing.

iamgoofballsecondfork commented:
dragon drilldongs and testing github bans with cheridan

Iamgoofball commented:
testing

Iamgoofball commented:
okay so cloneloss blobs:

they'll come with a buff to cloneloss healing chems

Iamgoofball commented:
testing

Iamgoofball committed:
nerfs the blobbernaut a bit

more fixes

forgot to commit storage blobs

fixes

Blob Mania 1.5: Storage Blobs and Blob Broadcast

nerfs dark matter

Reagent Blob tweaks and adjustments

Added new blob reagents!

Delimusca commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+
+datum/reagent/blob/lung_destroying_toxin
+	name = "Lung Destroying Toxin"
+	id = "boiling_oil"
```
duplicate ID?

Iamgoofball replied:
whoops

RemieRichards commented on code/game/gamemodes/blob/blobs/core.dm:
```diff
@@ -14,6 +14,9 @@
 	processing_objects.Add(src)
 	if(!overmind)
 		create_overmind(new_overmind)
+		color = overmind.blob_reagent_datum.color
```
I specifically removed this line as it causes runtimes, the line isn't needed as create_overmind() handles the colour now too. so please revert this back so it's the else chain

Atskadan commented:
is there a way to greyscale-color a sprite while still retaining actual colors?
if so you could make cores and nodes and stuff still yellow while the rest is whatever color

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+	if(method == TOUCH)
+		M.adjust_fire_stacks(2)
+		M.IgniteMob()
+		M << "You feel your skin burning!"
```
It's better descriptions time!
"The blob splashes you with burning oil!"

phil235 replied:
All these messages also needs span classes.

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+datum/reagent/blob/anti_oxygenation/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(20, OXY)
+		M << "You can't breathe!"
```
"The blob strikes you, and you feel dizzy and out of breath!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+		M.apply_damage(10, BURN)
+		M.adjust_fire_stacks(2)
+		M.IgniteMob()
+		M << "You feel your skin burning and melting!"
```
"The blob strikes you, and you feel your skin char and melt!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+datum/reagent/blob/explosive/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		if(prob(50))
+			M << "The blob explodes!"
```
"The blob strikes you, and its tendrils explode!"
This should probably happen a touch more often, also.

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+						if(S.magpulse)
+							continue
+					X.throw_at(pull)
+		M << "Everything flies at you!"
```
"You feel a thrum as the blob strikes you, and everything flies at you!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+	if(method == TOUCH)
+		M.acid_act(20,1,10)
+		M.apply_damage(10, BRUTE)
+		M << "You feel your skin and equipment melting off!"
```
"The blob's tendrils melt though your equipment!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+	if(method == TOUCH)
+		M.apply_damage(10, OXY)
+		M.apply_damage(10, TOX)
+		M << "You feel your lungs rotting!"
```
The blob strikes you, and your lungs feel heavy and weak!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+				end_dir = EAST
+			var/atom/throw_target = get_edge_target_turf(M, end_dir)
+			M.throw_at(throw_target, 200, 4)
+			M << "You are sent flying!"
```
"The blob slams into you, and sends you flying!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+				randmutg(M)
+			domutcheck(M, null)
+			updateappearance(M)
+		M << "You feel irradiated!"
```
"The blob strikes you, and your skin feels papery and everything hurts!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+	if(method == TOUCH)
+		M.apply_damage(30, STAMINA)
+		M.apply_damage(10, BRUTE)
+		M << "You feel incredibly tired!"
```
"The blob strikes you, and you feel winded and weak!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+datum/reagent/blob/skin_ripper/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(20, BRUTE)
+		M << "You feel your skin being ripped off!"
```
"The blob strikes you, and you feel your skin ripping and tearing off!"

Anonus commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,180 @@
+datum/reagent/blob/toxic_goop/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.apply_damage(20, TOX)
+		M << "You feel sick!"
```
"The blob strikes you, and you feel sick and nauseated!

Iamgoofball commented:
I love it, will change soon

Iamgoofball committed:
storage blob fixes, new descriptions, span classes

AlexanderUlanH commented:
Unless I missed some advantage that oxyloss blobs have, oxygen blobs are going to be much less powerful than any of the others.  Not only does their damage heal naturally over time, but there's dexplus, which sets oxyloss to 0 every tick, so if you load up with that, you're completely immune to the blob's attacks.

Iamgoofball commented:
hm, I'll look into buffing oxyblob

Anonus commented:
Dex+ is going to make oxyblobs way too weak and very dead even if you give them lexorin, because oxygen damage is stupidly easy to heal away.

iamgoofballsecondfork commented:
testing testing 123

RemieRichards commented:
@tgstation/commit-access review pls.

TZK13 commented:
I don't like how difficult it is to see blob structures like nodes and the core, can't you make those into overlays?

tkdrg commented:
I thought you were going to do something about oxyblob?

Iamgoofball commented:
my laptop took a huge shit again and it's undergoing repairs

i'll get the sprites modified soon and the overlay shit done and oxyblobs either removed/buffed a shitton

Iamgoofball committed:
Moves the core and node sprites to be overlays. Tweaks and adjustments to various reagents. Removes 2 of the reagents for being useless.

Iamgoofball committed:
whoops, didn't mean to forget to uncomment those

Iamgoofball commented:
sprites modified

stamina and oxygen blob removed

ready

Iamgoofball committed:
Enables children blobs to be able to split. Public playtesting can show whether this needs to be disabled.

Iamgoofball commented:
Enables children blobs to be able to split. Public playtesting can show whether this needs to be disabled.

Razharas commented on code/modules/reagents/Chemistry-Reagents/Blob-Reagents.dm:
```diff
@@ -0,0 +1,165 @@
+					step_away(X,pull)
+					step_away(X,pull)
+					step_away(X,pull)
+					step_away(X,pull)
```
This code piece is so bad i want to cri

Iamgoofball committed:
Fixes copypasta.

Razharas commented:
3/5 ill give this shit a try

RemieRichards commented:
@Iamgoofball >makes children blob able to split

oh lordy I prevented this for a reason! Guess it's time for the blob's victims to ghost and join in the /fun/

Iamgoofball commented:
It can be disabled if it turns into omega blob. I'd like to test run it enabled first.
